### ISSUE TITLE
bug: terminal doesn't respect End or Delete

### ISSUE THREAD

**Issue by 0x70b1a5** [uqbar-dao/nectar#431]
**Describe the bug**
End doesn't move your cursor, and Delete works like Backspace.

**To Reproduce**
1. enter some text
2. arrow key left some
3. hit Delete, notice Backspace
4. hit End, notice no cursor move 
**Expected behavior**
1. normal function of said keys

**Desktop (please complete the following information):**
```
            .-/+oossssoo+/-.               mogged@mcdonalds 
        `:+ssssssssssssssssss+:`           -------------- 
      -+ssssssssssssssssssyyssss+-         OS: Ubuntu 24.04 LTS x86_64 
    .ossssssssssssssssssdMMMNysssso.       Kernel: 6.8.0-36-generic 
   /ssssssssssshdmmNNmmyNMMMMhssssss/      Uptime: 4 hours, 21 mins 
  +ssssssssshmydMMMMMMMNddddyssssssss+     Packages: 2536 (dpkg), 46 (nix-user), 12 (flatpak), 30 (snap) 
 /sssssssshNMMMyhhyyyyhmNMMMNhssssssss/    Shell: bash 5.2.21 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Resolution: 1440x2560, 2560x1440 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   DE: GNOME 46.0 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM: Mutter 
ossyNMMMNyMMhsssssssssssssshmmmhssssssso   WM Theme: Adwaita 
+sssshhhyNMMNyssssssssssssyNMMMysssssss+   Theme: Yaru-blue-dark [GTK2/3] 
.ssssssssdMMMNhsssssssssshNMMMdssssssss.   Icons: Yaru-blue-dark [GTK2/3] 
 /sssssssshNMMMyhhyyyyhdNMMMNhssssssss/    Terminal: gnome-terminal 
  +sssssssssdmydMMMMMMMMddddyssssssss+     CPU: 13th Gen Intel i9-13900K (32) @ 5.500GHz 
   /ssssssssssshdmNNNNmyNMMMMhssssss/      GPU: NVIDIA GeForce RTX 3080 Ti 
    .ossssssssssssssssssdMMMNysssso.       Memory: 10256MiB / 64037MiB 
      -+sssssssssssssssssyyyssss+-
        `:+ssssssssssssssssss+:`                                   
            .-/+oossssoo+/-.                                       

```

**Comment by dr-frmr** [comment]
A previous issue specifically requested we treat "Delete" and "Backspace" in the same way. I don't care, as I only have a backspace key on my keyboard. I can definitely add proper behavior for "End".

**Comment by 0x70b1a5** [comment]
Do you mean #361 ? I don't think that was what that issue requested. Reading through it, Lyn seems to be asking for normal Delete functionality, not that Delete be treated like Backspace, but I could be wrong there - either way, by the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), the terminal should definitely have the normal Delete behavior.

**Comment by dr-frmr** [comment]
What is the delete key supposed to do vs. the backspace key? Sorry for being ignorant, I have always treated them as equivalent: delete the character to the right of the cursor.

**Comment by 0x70b1a5** [comment]
Oh my bad, I'm also just remembering now that Macs have their Backspace keys labelled as "Delete".
- Backspace deletes the character to the *left* of the cursor.
- Delete deletes the character to the *right* of the cursor (in a terminal, this would be the character highlighted by the cursor). On my 2017 mac, one can induce this behavior by pressing the Fn+"delete" (actually Backspace) keys.

**Comment by dr-frmr** [comment]
Huh ok. I can do that, sure

**Comment by dr-frmr** [comment]
fixed in #443 !
